deploy/docs/FluentdPersistence.md

```diff
@@ -0,0 +1,292 @@
+# Fluentd persistence
+
+Starting with `v2.0.0` we're using file-based buffer for Fluentd instead of less
+reliable in-memory buffer.
+
+The buffer configuration can be set in the `values.yaml` file under the `fluentd`
+key as follows:
+
+```yaml
+fluentd:
+  persistence:
+    enabled: true
+```
+
+When setting for Fluentd persistence is changed (enable or disable) it is required to recreate or delete existing Fluentd StatefulSet,
+as it is not possible to add/remove `volumeClaimTemplate` for StatefulSet.
+
+**Note:** The below commands are using `yq` in version `3.4.0` <= `x` < `4.0.0`.
+
+## Enabling Fluentd persistence
+
+To enable Fluentd persistence in existing collection modify `values.yaml` file under the `fluentd`
+key as follows:
+
+```yaml
+fluentd:
+  persistence:
+    enabled: true
+    ## If defined, storageClassName: <storageClass>
+    ## If set to "-", storageClassName: "", which disables dynamic provisioning
+    ## If undefined (the default) or set to null, no storageClassName spec is
+    ##   set, choosing the default provisioner (gp2 on AWS, standard on
+    ##   GKE, Azure & OpenStack)
+    ##
+    # storageClass: "-"
+    # annotations: {}
+    accessMode: ReadWriteOnce
+    size: 10Gi
+```
+
+Use one of following two strategies to prepare existing collection for enabling Fluentd persistence:
+
+- ### Enabling Fluentd persistence by recreating Fluentd StatefulSet
+
+  Recreating Fluentd StatefulSet with new `volumeClaimTemplate` may cause that logs and metrics
+  will not be available in the time of recreation. It usually shouldn't take more than several seconds.
```

Comment:
```suggestion
  In a heavy used clusters with high load of logs and metrics it might be possible that
  recreating Fluentd StatefulSet with new `volumeClaimTemplate` may cause logs and metrics
  being unavailable for the time of recreation. It usually shouldn't take more than several seconds.
```